services/src/main/java/org/fao/geonet/api/records/MetadataUtils.java

```diff
@@ -99,17 +90,312 @@ public class MetadataUtils {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(Geonet.SEARCH_ENGINE);
 
-    /**
-     * Constants for metadata origin:
-     *
-     *  - portal: the metadata is available in the current portal.
-     *  - catalog: the metadata is not available in the current portal, but is available in the local catalog.
-     *  - remote: the metadata is available in a remote resource, used for operatesOn resources.
-     */
-    private static final String ORIGIN_PORTAL = "portal";
-    private static final String ORIGIN_CATALOG = "catalog";
-    private static final String ORIGIN_REMOTE = "remote";
+    public static class RelatedTypeDetails {
+        private String query;
+        private Set<String> expectedRecords = new HashSet<>();
```

Comment:
What are the expected records? Records defined in the metadata relations?

Reply:
While working on this we were also discussing a way to track bad relations (eg. if a record is removed but is still linked) so we may have deadlinks at some point. The search is built in most case on top of a list of UUIDs to search for and if user is admin expectedRecords is supposed to match the number of UUID to found. We did not complete that part as the link analysis partially analyse that aspect too (but also have drawbacks eg. private records can not be checked). So further work is required in that area.